### ISSUE TITLE
Backport fix for `include/__clang_cuda_math.h:39:25: warning: extra tokens at end of #ifdef directive [-Wextra-tokens] ` in LLVM 17

### ISSUE THREAD

**Issue by joker-eph** [llvm/llvm-project#64718]
c49142e4f5c8 fixed this warning:

```
include/__clang_cuda_math.h:39:25: warning: extra tokens at end of #ifdef directive [-Wextra-tokens]
```                            ^
